supplementary/envelopes_analysis/field_envelopes.py

```diff
@@ -1,163 +1,18 @@
-        dt = 2*np.pi/omega/50
-        t_ft = np.arange(self.tmin - 20*self.field_fwhm, self.tmax + 20*self.field_fwhm, dt)
-        field = self.calc_field_envelope(t_ft)*self.field_cos(t_ft)
-        self.field_ft = np.abs(np.fft.rfft(field))  # FT
-        self.field_ft /= np.max(self.field_ft)  # normalizing
-        self.field_ft_omega = 2*np.pi*np.fft.rfftfreq(len(t_ft), dt)
-
-        self.field_calculated = True
-
-    def pulse_wigner(self, tprime, de):
-        """
-        Wigner transform of the pulse envelope as originally propposed by Martínez-Mesa and Saalfrank.
-        :param tprime: time at which the molecule is excited (a.u.)
-        :param de: excitation energy (a.u.)
-        :return:
-        """
-
-        if not self.field_calculated:
-            print(f"ERROR: Input data not read yet. Please first use 'read_input_data()'!")
-            exit(1)
-
-        # attempt for an adaptive integration step according to the oscillation
-        # however, for plotting on a grid it doesn't work as the edges have dE-omega = 12 or more so I needed to add a
-        # thresh with omega. Implementing this for the sampling shouldn't be a problem because I will have only relevant
-        # excitation energies.
-        loc_omega = self.field_omega + self.field_lchirp*tprime
-        if de != loc_omega:
-            T = 2*np.pi/(np.min([np.abs(de - loc_omega), loc_omega]))
-        else:  # in case they are equal, I cannot divide by 0
-            T = 2*np.pi/loc_omega
-        ds = np.min([T/50, self.field_fwhm/500])
-
-        # integration ranges for different pulse envelopes
-        # ideally, we would integrate from -infinity to infinity, yet this is not very computationally efficient
-        # empirically, it was found out that efficient integration varies for different pulses
-        if self.field_envelope_type == 'gauss':
-            factor = 7.5
-        elif self.field_envelope_type == 'lorentz':
-            factor = 50
-        elif self.field_envelope_type == 'sech':
-            factor = 20
-        elif self.field_envelope_type == 'sin':
-            factor = 3
-        elif self.field_envelope_type == 'sin2':
-            factor = 4
-
-        # instead of calculating the complex integral int_{-inf}^{inf}[E(t+s/2)E(t-s/2)exp(i(w-de)s)]ds we use the
-        # properties of even and odd fucntions and calculate 2*int_{0}^{inf}[E(t+s/2)E(t-s/2)cos((w-de)s)]ds
-        s = np.arange(0, factor*self.field_fwhm, step=ds)
-        cos = np.cos((de/self.hbar - loc_omega)*s)
-        integral = 2*np.trapz(x=s, y=cos*self.calc_field_envelope(tprime + s/2)*self.calc_field_envelope(tprime - s/2))
-
-        W = 1/2/np.pi/self.hbar*integral
-
-        return W
-
+from promdens.promdens import LaserPulse, InitialConditions, ENVELOPE_TYPES
 
 # input data
 fwhm = 10  # in fs
 omega = 1  # in a.u.
 lchirp = 0  # in a.u.
 t0 = 0  # in fs
-envelope_types = ['sin', 'sin2', 'gauss', 'sech', 'lorentz']
```

Comment:
Let's leave this list here, it's much easier to understand that the sorting below.